### ISSUE TITLE
Moq1203 false positive: delegate-based ReturnsAsync not recognized as return value specification

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#910]
### Description

Moq1203 incorrectly flags `Setup` calls as missing a return value specification when `ReturnsAsync` is used with a delegate parameter (e.g., `ReturnsAsync((MyValue val) => val)`). The scalar form `ReturnsAsync(value)` is recognized correctly.

### Steps To Reproduce

```csharp
public record MyValue;

public interface IDatabase
{
    Task<MyValue> SaveAsync(MyValue value);
}

public class UnitTest
{
    [Test]
    public async Task Test1()
    {
        var databaseMock = new Mock<IDatabase>(MockBehavior.Strict);

        // Incorrectly flags Moq1203: "Method setup should specify a return value"
        databaseMock
            .Setup(mock => mock.SaveAsync(It.IsAny<MyValue>()))
            .ReturnsAsync((MyValue val) => val);
    }
}
```

### Expected Behavior

No diagnostic. `ReturnsAsync((MyValue val) => val)` is a valid return value specification using the delegate overload.

### Actual Behavior

Moq1203 warning: "Method setup for 'IDatabase.SaveAsync(MyValue)' should specify a return value"

### Root Cause

`HasReturnValueSpecification` in `MethodSetupShouldSpecifyReturnValueAnalyzer` walks the method chain and calls `semanticModel.GetSymbolInfo(memberAccess)` on each `MemberAccessExpressionSyntax`. When `ReturnsAsync` is called with a delegate lambda, Roslyn may fail to resolve the specific overload, causing `HasReturnValueSymbol` to return false even though `ReturnsAsync` is present in the chain.

The scalar overloads (`ReturnsAsync(TResult)`) resolve correctly because overload resolution is simpler.

### Affected Overloads

Moq's `ReturnsExtensions` defines delegate-based overloads:
- `ReturnsAsync<T, TResult>(Func<T, TResult>)`
- `ReturnsAsync<T1, T2, TResult>(Func<T1, T2, TResult>)`
- etc.

These are not recognized by the analyzer.

### Regression Tests

PR #907 added test cases documenting this false positive in `CustomReturnTypeMissingReturnValueTestData`. When this issue is fixed, move those cases to `CustomReturnTypeTestData` and remove the `{|Moq1203:...|}` diagnostic markup.

### Environment

- Moq.Analyzers: 0.5.0-alpha (main branch)
- Reported by: @DamienCassou in #849 (comment: https://github.com/rjmurillo/moq.analyzers/issues/849#issuecomment-3925720443)
